language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +177,146 @@
+        Optional<MetadataNode> metadata = typeDefinitionNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
         return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
                 range, range, isDeprecated, children, this.context));
     }
 
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        String name = moduleVariableDeclarationNode.typedBindingPattern().bindingPattern().toSourceCode();
+        SymbolKind symbolKind = SymbolKind.Variable;
+        Range range = DocumentSymbolUtil.generateNodeRange(moduleVariableDeclarationNode);
+        Optional<MetadataNode> metadata = moduleVariableDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ConstantDeclarationNode constantDeclarationNode) {
+        String name = constantDeclarationNode.variableName().text();
+        SymbolKind symbolKind = SymbolKind.Constant;
+        Range range = DocumentSymbolUtil.generateNodeRange(constantDeclarationNode);
+        Optional<MetadataNode> metadata = constantDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(EnumDeclarationNode enumDeclarationNode) {
+        String name = enumDeclarationNode.identifier().text();
+        SymbolKind symbolKind = SymbolKind.Enum;
+        Range range = DocumentSymbolUtil.generateNodeRange(enumDeclarationNode);
+        Optional<MetadataNode> metadata = enumDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleXMLNamespaceDeclarationNode moduleXMLNamespaceDeclarationNode) {
+        String name = SyntaxKind.XMLNS_KEYWORD.stringValue() + " "
+                + moduleXMLNamespaceDeclarationNode.namespaceuri().toSourceCode();
+        Optional<IdentifierToken> prefix = moduleXMLNamespaceDeclarationNode.namespacePrefix();
+        if (prefix.isPresent()) {
+            name = name + " as " + prefix.get();
```

Comment:
If we have the prefix, let's just add the prefix only, because we refer the symbol with the prefix itself